src/output-json-dns.c

```diff
@@ -465,6 +465,22 @@ static void OutputAnswer(LogDnsLogThread *aft, json_t *djs,
     /* id */
     json_object_set_new(js, "id", json_integer(tx->tx_id));
 
+    /* dns */
+    char flags[7] = "";
+    snprintf(flags, sizeof(flags), "%4x", tx->flags);
+    json_object_set_new(js, "flags", json_string(flags));
```

Comment:
If its a string, of a hex number, I wonder if it should include the leading 0x? I'm trying to find some existing JSON precedent. But as it is a serialization format I think it should just be logged as an integer, the viewer can convert it back to hex?

Does our JSON have existing precedent of logging hex as a string without the leading 0x?

Reply:
We do this in flow/netflow records, for tcp flags set. The only feedback I've ever had on that field was negative though. But perhaps it's a documentation issue.

Reply:
In previous commit (PR 2920), I was adding the 0x. However, since for TCP netflow, the 0x is not print in the json, I have proposed for this current PR to do the same: for both, no 0x in keep coherency and same philosophy for both (DNS or Netflow TCP flag): no 0x for the hexa output.

Reply:
Yes agreed, I need to update documentation (eve-json-format.rst) file with this feature: explain new fields, and provide the 0x notice in this section.